app/Providers/FortifyServiceProvider.php

```
@@ -127,13 +127,19 @@ public function boot(): void
+            return Limit::perMinute(5)->by($realIp);
         });
 
         RateLimiter::for('login', function (Request $request) {
             $email = (string) $request->email;
+            // Use email + real client IP (not spoofable forwarded headers)
+            // server('REMOTE_ADDR') gives the actual connecting IP before proxy headers
+            $realIp = $request->server('REMOTE_ADDR') ?? $request->ip();
 
-            return Limit::perMinute(5)->by($email.$request->ip());
+            return Limit::perMinute(5)->by($email.$realIp);
         });
```

PHPMD (2.15.0)

Avoid using static access to class '\Illuminate\Support\Facades\RateLimiter' in method 'boot'. (undefined)

(StaticAccess)